### PR TITLE
Adding in doctests + renaming x86/x64 to i386/amd64

### PR DESCRIPTION
**Pull request by eQu1NoX**
1. Added in basic doctests for i386 and amd64.
2. Renamed x86/x64 to i386/amd64.

### CODE REVIEW

**Comment by zachriggle** [comment]
You'll need to add a `.rst` file to the `docs/source/` tree in order for `sphinx` to pick up the change, and add it to `docs/source/index.rst`.

**Comment by zachriggle** [comment]
All of the `struct.xxx` can be replaced with the pwntools versions from `pwnlib.util.packing`, which are `context`-aware and should always ensure the right thing happens.

File: pwnlib/srop.py [hunk 1]
```diff
@@ -87,8 +107,12 @@ def _set_regvalue_x86(self, reg, val):
...
-        if self.arch == "x86":
+        if self.arch == "i386":
             assert len(frame_contents) == len(_registers_32) * 4
-        elif self.arch == "x64":
+        elif self.arch == "amd64":
             assert len(frame_contents) == len(_registers_64) * 8
         return frame_contents
+
+if __name__ == '__main__':
+    import doctest
```
Comment (zachriggle) [inline pwnlib/srop.py]: Drop this, it gets picked up automatically by the doctest runner when there's an RST file (see my comment)

File: pwnlib/srop.py [hunk 2]
```diff
@@ -87,8 +107,12 @@ def _set_regvalue_x86(self, reg, val):
 
     def get_frame(self):
         frame_contents = ''.join(self.frame)
-        if self.arch == "x86":
+        if self.arch == "i386":
```
Comment (zachriggle) [inline pwnlib/srop.py]: Most of these statements can be dropped with a dict lookup. 

```
_reg_pos_mapping_i386 = ...
_reg_pos_mapping_amd64 = ...
```

vs:

```
_reg_pos_mapping = {
 'i386': ...,
 'amd64': ...
}
```

**Comment by zachriggle** [comment]
I'll pull this in if you can squash all those into one commit and replace the super-long lines with `unpack_many` :smile: 

Example:

```
>>> assert len(frame) == 248
>>> unpack_many(frame)
[0, 0, 0, 0, 0, 0, 0, 0, 6295552, 7, 4096, 125, 0, 0, 0, 115, 0, 0, 123, 0]
```

```
>>> assert len(frame) == 248
>>> print repr(frame)
'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10`\x00\x00\x00\x00\x00\x00\x10\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x003\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
```

**Comment by eQu1NoX** [comment]
Hi 

I've squashed the commits in and replaced the long lines with unpack_many :)

**Pull request merged**